File: go/621. Task Scheduler.go
func leastInterval(tasks []byte, n int) int {
	freq := [26]int{}
	for _, t := range tasks {
		freq[t-'A']++
	}
	sort.Ints(freq[:])
	maxFreq := freq[25] - 1
	idleSlots := maxFreq * n
	for i := 24; i >= 0 && freq[i] > 0; i-- {
		idleSlots -= min(maxFreq, freq[i])
	}
	return max(len(tasks), len(tasks)+idleSlots)
}

File: python/621. Task Scheduler.py
class Solution:
    def leastInterval(self, tasks: List[str], n: int) -> int:
        freq = [0] * 26
        for t in tasks:
            freq[ord(t) - ord('A')] += 1
        freq.sort()
        f_max = freq.pop()
        idle_time = (f_max - 1) * n
        while freq and idle_time > 0:
            idle_time -= min(f_max - 1, freq.pop())
        return idle_time + len(tasks) if idle_time > 0 else len(tasks)
